spec/repository_spec.rb

```diff
@@ -26,3 +26,17 @@
     expect(EXTRA_FILES).to eq([])
   end
 end
+
+describe "The file \"renamed-styles.json\"" do
+  it "must be valid JSON" do
+    expect(JSON_IS_VALID).to be true
+  end
```

Comment:
Good questions. Validation is certainly a factor, but I'd suggest to disable it temporarily to see what kind of performance hit we're actually talking about. The reason why this is a little complicated is that, yes, the styles are in memory, but that is actually the Ruby representation of each style; we could theoretically gain some speed if we read each XML file just once validate and parse the style right away -- it's not straightforward so it would be good to know exactly how much this would be worth (though it is probably a considerable amount of time).